make clang-tidy runs on header files of library

When I run `make clang-tidy`, it changes header files of libraries like OpenSSL or PCRE. It looks like we need to some tweaks with [`-header-filter`](https://github.com/apache/trafficserver/blob/0933b163b063cfe162864ee287d567197a67f429/build/tidy.mk#L17)

e.g.
```
/opt/openssl-1.1.1/include/openssl/ossl_typ.h:13:10: warning: inclusion of deprecated C++ header 'limits.h'; consider using 'climits' instead [modernize-deprecated-headers]
#include <limits.h>
         ^~~~~~~~~~
         <climits>
```

--- a/.clang-tidy
+++ b/.clang-tidy
@@ -1,1 +1,2 @@
 Checks: -*,modernize-loop-convert,modernize-use-bool-literals,modernize-deprecated-headers,performance-unnecessary-value-param,performance-faster-string-find,modernize-raw-string-literal,modernize-redundant-void-arg,modernize-use-nullptr,modernize-use-override
+HeaderFilterRegex: (?!(\/openssl\/|\/pcre\/|\/lib\/yaml\/)).*

--- a/build/tidy.mk
+++ b/build/tidy.mk
@@ -14,7 +14,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-Clang_Tidy_Options = -fix -fix-errors -header-filter=.*
+Clang_Tidy_Options = -fix -fix-errors
 
 # Sort the filenames to remove duplicates, then filter to retain
 # just the C and C++ sources so we don't pick up lex and yacc files

